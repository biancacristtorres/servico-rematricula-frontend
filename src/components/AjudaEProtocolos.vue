<template>
  <router-link id="url-fresh-desk" v-if="usuarioEstaLogado" class='link-flat' :to="'Ajuda'" target="_blank">
    <v-btn v-if="$vuetify.breakpoint.smAndDown"  icon flat>
      <v-icon >
        help_outline
      </v-icon>
    </v-btn>

    <v-btn v-else flat>
      Ajuda e protocolos
    </v-btn>
  </router-link>
</template>

<style lang="stylus" scoped>
.btnAjudaEProtocolos
  width: 184px
  text-transform: none
  
.link-flat
  text-decoration: none
</style>

<script lang='ts'>
import { Component, Vue } from 'vue-property-decorator';
import { StoreNamespaces } from '@/store';
import { namespace } from 'vuex-class';
import { AutorizacaoGetterTypes } from 'componente-frontend-core/store/autorizacao/getters';

const autorizacaoNamespace = namespace(StoreNamespaces.AUTORIZACAO);
const marca = namespace(StoreNamespaces.MARCA);

@Component({})
export default class PessoaLogada extends Vue {
  @autorizacaoNamespace.Getter(AutorizacaoGetterTypes.USUARIO_ESTA_LOGADO)
  public usuarioEstaLogado!: boolean;

  public get deveExibirComponente() {
    return this.usuarioEstaLogado;
  }
}
</script>