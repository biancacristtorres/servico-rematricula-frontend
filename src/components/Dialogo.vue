<template>
  <v-dialog id="dialogo" v-model="opcoes.exibir" persistent :max-width="500">
    <v-card id="dialogo-conteudo">
      <v-card-title id="titulo" class="headline" primary-title>
        {{ opcoes.titulo }}
      </v-card-title>

      <v-card-text id="texto" v-html="opcoes.texto">
        {{opcoes.texto}}
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>
      
        <v-btn id="rejeitar" color="red" flat v-if="opcoes.textoRejeitar"  @click.native="aoRejeitar">
          {{ opcoes.textoRejeitar }}
        </v-btn>

        <v-btn id="confirmar" color="primary" flat v-if="opcoes.textoConfirmar" @click.native="aoConfirmar">
          {{ opcoes.textoConfirmar }}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script lang='ts'>
import { Component, Prop, Vue } from 'vue-property-decorator';
import { OpcoesDeDialogo } from '../model/OpcoesDeDialogo';

@Component({})
export default class Dialogo extends Vue {

  @Prop()
  private opcoes!: OpcoesDeDialogo;

  public aoConfirmar() {
    this.$emit('confirmar');
  }

  public aoRejeitar() {
    this.$emit('rejeitar');
  }
}
</script>