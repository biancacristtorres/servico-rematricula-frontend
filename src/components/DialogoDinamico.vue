<template>
  <v-dialog v-model="opcoes.exibir" scrollable persistent :max-width="500">
    <v-card id="dialogo-dinamico" ref="dialogo-dinamico">
      <v-card-title class="headline" primary-title>
          {{ opcoes.titulo }}
      </v-card-title>

      <v-card-text style="max-height: 300px">
        <component :is="componenteDinamico" :model="opcoes.modelDoComponente">
        </component>
      </v-card-text>

      <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn id="cancelar" 
              color="red" 
              flat 
              v-if="opcoes.textoCancelar"
              @click.native="aoRejeitar">
            {{ opcoes.textoCancelar }}
          </v-btn>

          <v-btn id="confirmar" 
            color="primary"
            flat
            v-if="opcoes.textoConfirmar"
            @click.native="aoConfirmar">
            {{ opcoes.textoConfirmar }}
          </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script lang='ts'>
import { Component, Prop, Vue } from 'vue-property-decorator';
import { OpcoesDeDialogoDinamico } from '@/model/OpcoesDeDialogoDinamico';

@Component({})
export default class DialogoDinamico extends Vue {

  @Prop()
  public opcoes!: OpcoesDeDialogoDinamico;

  private get componenteDinamico(): any {
    return () => import(`@/components/${ this.opcoes.componente }`);
  }

  public aoConfirmar() {
    this.$emit('confirmar');
  }

  public aoRejeitar() {
    this.$emit('rejeitar');
  }

}
</script>
