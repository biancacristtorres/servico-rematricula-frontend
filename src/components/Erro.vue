<template>
  <v-container class="grey--text" fluid>
    <v-layout wrap align-center row reverse justify-center fill-height>

      <v-flex class="text-xs-center pt-5" xs12 md6>
        <v-layout wrap row fill-height
          :justify-center="$vuetify.breakpoint.smAndDown" 
          :justify-start="$vuetify.breakpoint.mdAndUp">
          <v-flex class="coluna" xs12>
            <v-img id="imagem-erro" class="text-xs-center coluna" :src="getImgUrl"/>
          </v-flex>
        </v-layout>

      </v-flex>

      <v-flex xs12 md6>
        <v-layout wrap row fill-height
          :justify-center="$vuetify.breakpoint.smAndDown"
          :justify-end="$vuetify.breakpoint.mdAndUp" >

          <v-flex class="coluna" xs12>
            <p id="titulo-erro" class="display-2 text-xs-left">
              {{ titulo }}
            </p>
          </v-flex>

          <v-flex class="coluna" xs12>
            <div id="texto-erro" class="pt-3 texto subheading text-xs-left">
              {{ texto }}
            </div>
          </v-flex>

          <v-flex v-if="deveRedirecionarParaOSOL" 
            class="coluna" 
            :class="classeContainerBotao"
            xs12>
            <v-btn id="retornar-ao-sol" :href="linkDoSOL" class="mt-3 ml-0 btn-retornar-ao-sol" color="primary">
              Retornar ao SOL
            </v-btn>
          </v-flex>
        </v-layout>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Instituicao } from '@/common/enum/Instituicao';
import { Marca } from '@/model/Marca';
import { StoreNamespaces } from '@/store';
import { MarcaGetterTypes } from '@/store/marca/getters';
import { InformacoesAluno } from '@/model/InformacoesAluno';
import { namespace } from 'vuex-class';

const aluno = namespace(StoreNamespaces.ALUNO);
const marca = namespace(StoreNamespaces.MARCA);

@Component({})
export default class Erro extends Vue {

  public get linkDoSOL(): string {
    return this.obterLinkDoSol(this.informacoesAluno.codigoInstituicao);
  }

  public get getImgUrl() {
    try {
      return require(`./assets/erro/${ this.imagem }.svg`);
    } catch {
      return '';
    }
  }

  public get classeContainerBotao() {
    return this.$vuetify.breakpoint.smAndDown ? 'text-xs-center' : '';
  }

  @aluno.State
  public informacoesAluno!: InformacoesAluno;

  @marca.Getter(MarcaGetterTypes.OBTER_LINK_DO_SOL)
  public obterLinkDoSol!: (instituicao: Instituicao) => string;

  @Prop()
  private titulo!: string;

  @Prop()
  private texto!: string;

  @Prop()
  private imagem!: string;

  @Prop({ default: true })
  private deveRedirecionarParaOSOL!: boolean;

}
</script>

<style lang="stylus" scoped>
.btn-retornar-ao-sol
  min-width: 280px
.coluna
  max-width: 416px
</style>