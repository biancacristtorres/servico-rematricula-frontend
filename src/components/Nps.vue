<template>
<v-snackbar :value="true" :timeout="0" bottom color="secondary">
  Avalie sua rematrícula
  <v-btn id="clique-aqui" @click="abrir" flat>
    Clique aqui
    <v-dialog @update:returnValue="limparNps" v-model="mostrarNPS" width="432px">
      <v-card>
        <v-img :src="require('./assets/nps/word-of-mouth.png')"></v-img>
        <div class="pl-4 pr-4 pt-2 pb-2">
          <p class="text-xs-center pt-2 title text--grey">{{ texto }}</p>
          <v-layout class="text-xs-center" row justify-space-between>
            <v-flex xs2>
              <v-img :style="`opacity: ${ avaliacaoRuim ? 1 : 0.4 }`" :src="imagemNps('face-bad')"></v-img>
            </v-flex>
            <v-flex xs2>
              <v-img :style="`opacity: ${ avaliacaoMediana ? 1 : 0.4 }`" :src="imagemNps('face-indifference')"></v-img>
            </v-flex>
            <v-flex xs2>
              <v-img :style="`opacity: ${ avaliacaoBoa ? 1 : 0.4}`" :src="imagemNps('face-loveit')"></v-img>
            </v-flex>
          </v-layout>
          <v-radio-group :value="nps" @change="alterarNPS" row>
            <v-container grid-list-md pa-0>
              <v-layout row wrap justify-space-around>
                <v-flex v-for="i in 10" :key="`1${i}`" xs1>
                  <v-radio color="primary" :ripple="false" :value="i"></v-radio>
                  <span class="subheading radio-label">{{ i }}</span>
                </v-flex>
              </v-layout>
            </v-container>
          </v-radio-group>
          <v-textarea v-model="observacoes" class="nps-observacoes pa-3" name="nps-observacoes" placeholder="Comentário (opcional)" rows="4"></v-textarea>
          <p class="text-xs-center pt-3">
            <v-btn :disabled="nps < 1" color="primary" class="botao-enviar" @click="votar">Enviar</v-btn>
          </p>
        </div>
      </v-card>
    </v-dialog>
  </v-btn>
  </v-snackbar>
</template>

<style lang="stylus" scoped>
.nps-observacoes
  border-radius: 2px
  border: solid 1px #dedede

.radio-label
  padding-left: 7px

.botao-enviar
  width: 192px
  color: primary
</style>

<script lang='ts'>
import { Component, Prop, Vue } from 'vue-property-decorator';
import NpsPageViewEvent from '@/model/analytics/NpsPageViewEvent';
import NpsEnvioEvent, { NpsEnvioCategory } from '@/model/analytics/NpsEnvioEvent';
import { OpcoesDeDialogo } from '@/model/OpcoesDeDialogo';

@Component({})
export default class Nps extends Vue {
  public mostrarNPS: boolean = false;
  public nps: number = 0;
  public observacoes: string = '';

  public texto: string = `Você indicaria a nova rematrícula para um(a) amigo(a)?`;
  public imagemNps(nomeImagem: string) {
    return require(`./assets/nps/${nomeImagem}.svg`);
  }

  public get avaliacaoRuim() {
    return this.nps <= 4 && this.nps !== 0;
  }

  public get avaliacaoMediana() {
    return this.nps > 4 && this.nps < 9;
  }

  public get avaliacaoBoa() {
    return this.nps >= 9;
  }

  public abrir() {
      this.$analytics.track(new NpsPageViewEvent(false));
      this.mostrarNPS = true;
  }

  public async votar() {
    this.$analytics.track(new NpsEnvioEvent(NpsEnvioCategory.COMENTARIO, this.nps, this.observacoes));
    this.$analytics.track(new NpsEnvioEvent(NpsEnvioCategory.NOTA, this.nps, this.observacoes));

    this.limparNps();

    this.exibirModalFeedbackEnvio('Avaliação Enviada',
        'Sua avaliação foi enviada com sucesso. Muito obrigado pelo Feedback! :)',
        'OK');

    this.$analytics.track(new NpsPageViewEvent(true));
  }

  public limparNps() {
    this.mostrarNPS = false;
    this.nps = 0;
    this.observacoes = '';
  }

  public exibirModalFeedbackEnvio(titulo: string, texto: string, botao: string) {
    const opcoes = new OpcoesDeDialogo (
      titulo,
      texto,
      botao,
      '',
    );

    this.$dialogo.abrir(opcoes);
  }

  public alterarNPS(valor: number) {
    this.nps = valor;
    this.$analytics.track(new NpsEnvioEvent(NpsEnvioCategory.CLIQUE, this.nps, ''));
  }
}
</script>