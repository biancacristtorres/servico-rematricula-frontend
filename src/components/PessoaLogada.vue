<template>
  <v-toolbar-items  id="pessoa-logada-container" v-if="existeUsuarioLogado">
    <input type="hidden" id="registro-academico" :value="registroAcademico">

    <v-menu offset-y>
      <v-btn v-if="$vuetify.breakpoint.smAndDown"  slot="activator" icon flat>
        <v-icon >
          account_circle
        </v-icon>
      </v-btn>

      <v-btn v-else slot="activator" flat>
          <span id="nome-pessoa">
            {{ nomeSocial }}
          </span>
          <v-icon right dark>
            arrow_drop_down
          </v-icon>
      </v-btn>

      <v-list light>
        <v-list-tile  @click="logoff">
          <v-list-tile-title id="logoff">
            Sair
          </v-list-tile-title>
        </v-list-tile>
      </v-list>
    </v-menu>

  </v-toolbar-items>
</template>

<script lang='ts'>
import UsuarioIdentity from 'componente-frontend-core/model/UsuarioIdentity';
import { Instituicao } from '@/common/enum/Instituicao';
import { Marca } from '@/model/Marca';
import { Component, Prop, Vue } from 'vue-property-decorator';
import { StoreNamespaces } from '@/store';
import { GlobalActionTypes } from '@/store/actions';
import { namespace, Action } from 'vuex-class';
import { AutorizacaoActionTypes } from 'componente-frontend-core/store/autorizacao/actions';
import { MarcaGetterTypes } from '@/store/marca/getters';
import { InformacoesAluno } from '@/model/InformacoesAluno';
import { FreshChat } from '@/common/utils/FreshChat';

const autorizacaoNamespace = namespace(StoreNamespaces.AUTORIZACAO);
const marca = namespace(StoreNamespaces.MARCA);
const aluno = namespace(StoreNamespaces.ALUNO);


@Component({})
export default class PessoaLogada extends Vue {

  @autorizacaoNamespace.State
  public usuarioLogado!: UsuarioIdentity;

  @aluno.State
  public informacoesAluno!: InformacoesAluno;

  public linkDoSol!: string;

  @autorizacaoNamespace.Action(AutorizacaoActionTypes.FAZER_LOGOUT)
  public fazerLogout!: () => Promise<void>;

  @Action(GlobalActionTypes.LIMPAR_STORE)
  public limparStore!: () => Promise<void>;

  @marca.Getter(MarcaGetterTypes.OBTER_LINK_DO_SOL)
  public obterLinkDoSol!: (instituicao: Instituicao) => string;

  public get existeUsuarioLogado() {
    return this.usuarioLogado !== null;
  }

  public get nomeSocial(): string {
    return this.informacoesAluno ? this.informacoesAluno.nomeSocial! : '';
  }

  public get registroAcademico(): string {
    return this.informacoesAluno ? this.informacoesAluno.numeroMatricula! : '';
  }

  public logoff() {
    this.linkDoSol = this.obterLinkDoSol(
      this.informacoesAluno.codigoInstituicao,
    );

    this.fazerLogout().then(this.aoFazerLogoff);
  }

  public async aoFazerLogoff() {
    await this.limparStore();
    FreshChat.limparUsuario();
    window.location.assign(this.linkDoSol);
  }
}
</script>