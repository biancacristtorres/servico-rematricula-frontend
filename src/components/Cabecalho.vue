<template>
  <v-toolbar color="primary" dark >
    <template v-if="deveExibirVoltar">
      <v-btn id="link-para-voltar" @click.native="voltar" icon>
        <v-icon>
          arrow_back
        </v-icon>
      </v-btn>
    </template>

    <v-toolbar-title id="titulo-cabecalho">
      {{ titulo }}
    </v-toolbar-title>

    <v-spacer></v-spacer>
    
    <ajuda-e-protocolos v-if="deveExibir" id="ajuda-e-protocolos">
    </ajuda-e-protocolos>

    <pessoa-logada v-if="deveExibir" id="pessoa-logada">
    </pessoa-logada>
  </v-toolbar>
</template>

<script lang="ts">
import { Component, Prop, Watch, Vue } from 'vue-property-decorator';
import { Action, State, namespace } from 'vuex-class';
import { StoreNamespaces } from '@/store';
import { GlobalActionTypes } from '@/store/actions';
import { AlteracaoGetterTypes } from '@/store/alteracao/getters';
import { OpcoesDeDialogo } from '@/model/OpcoesDeDialogo';
import { RotasSemValidacao } from '@/config/router/RotasSemValidacao';
import AjudaEProtocolos from '@/components/AjudaEProtocolos.vue';
import PessoaLogada from '@/components/PessoaLogada.vue';

const alteracao = namespace(StoreNamespaces.ALTERACAO);

@Component({components: { AjudaEProtocolos, PessoaLogada }})
export default class Cabecalho extends Vue {

  public get deveExibirVoltar(): boolean {
    return Boolean(this.linkParaVoltar);
  }

  @Action(GlobalActionTypes.LIMPAR_LINK_PARA_VOLTAR)
  public limparLinkParaVoltar!: () => Promise<void>;

  @alteracao.Getter(AlteracaoGetterTypes.POSSUI_ALTERACAO)
  public possuiAlteracao!: boolean;

  @Prop()
  private titulo!: string;

  @State
  private linkParaVoltar!: string;

  public async voltar(): Promise<void> {
    if (await this.deveRedirecionarParaOResumo()) {
      this.redirecionarParaOResumo();
    }
  }

  private async deveRedirecionarParaOResumo(): Promise<boolean> {
    return this.possuiAlteracao ?
      await this.usuarioQuerPerderAlteracoes() :
      true;
  }

  private async usuarioQuerPerderAlteracoes(): Promise<boolean>  {
    const opcoes = new OpcoesDeDialogo(
      'OPS! Você não salvou',
      'Você não salvou suas alterações, tem certeza que deseja sair assim mesmo?',
      'SIM',
      'NÃO',
    );

    try {
      await this.$dialogo.abrir(opcoes);
      return true;
    } catch (error) {
      return false;
    }
  }

  private redirecionarParaOResumo(): void {
    this.$router.push(this.linkParaVoltar);
  }

  private get deveExibir(): boolean {
    return !this.estaNaTelaDeErroDeSistema;
  }

  private get estaNaTelaDeErroDeSistema(): boolean {
    const fullPath = this.$route.fullPath.toLowerCase();
    const erroDeSistema = RotasSemValidacao.ErroDeSistema.toLowerCase();
    return fullPath.includes(erroDeSistema);
  }

  @Watch('$route', { immediate: true, deep: true })
  private aoMudarURL() {
    this.limparLinkParaVoltar();
  }
}
</script>
