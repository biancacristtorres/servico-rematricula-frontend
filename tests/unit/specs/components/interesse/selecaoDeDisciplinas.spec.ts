import { SelecaoDeDisciplinasFactory } from '../../../factory/components/SelecaoDeDisciplinasFactory';

describe('Componente de Interesse', () => {
  const factory = new SelecaoDeDisciplinasFactory();

  beforeEach(() => {
    factory.prepararContainerDeInjecaoDaApi();
    factory.dadoQueApiRetorneTresDisciplinas();
  });

  afterEach(() => {
    factory.restaurarContainerDeInjecaoDaApi();
  });

  it('snapshot', async (done) => {
    await factory.montarComponente();

    expect(factory.componente)
      .toMatchSnapshot();

    done();
  });
});
