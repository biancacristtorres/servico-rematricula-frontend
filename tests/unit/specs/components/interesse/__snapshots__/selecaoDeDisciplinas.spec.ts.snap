// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Interesse snapshot 1`] = `
<div class="container">
  <div class="layout">
    <div class="flex xs12">
      <div class="v-card v-card--flat theme--light">
        <div class="v-card__title text-xs-left pb-0 body-2 v-card__title--primary">
          Tem interesse em fazer alguma dessas disciplinas neste módulo?
        </div>
        <div class="v-card__text py-0 px-0">
          <div class="container px-3 pt-0 fluid">
            <div class="v-input disciplinas-pendentes v-input--selection-controls v-input--checkbox v-input--hide-details theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="disciplina-1" role="checkbox" type="checkbox" value="[object Object]">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">
                    <div class="body-1 black--text">
                      Cálculo I
                    </div>
                  </label>
                </div>
              </div>
            </div>
            <div class="v-input disciplinas-pendentes v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="true" id="disciplina-2" role="checkbox" type="checkbox" value="[object Object]">
                    <div class="v-input--selection-controls__ripple primary--text"></div><i aria-hidden="true" class="v-icon material-icons theme--light primary--text">check_box</i>
                  </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">
                    <div class="body-1 black--text">
                      Física II
                    </div>
                  </label>
                </div>
              </div>
            </div>
            <div class="v-input disciplinas-pendentes v-input--selection-controls v-input--checkbox v-input--hide-details theme--light">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="disciplina-3" role="checkbox" type="checkbox" value="[object Object]">
                    <div class="v-input--selection-controls__ripple"></div><i aria-hidden="true" class="v-icon material-icons theme--light">check_box_outline_blank</i>
                  </div><label aria-hidden="true" class="v-label theme--light" style="left: 0px; position: relative;">
                    <div class="body-1 black--text">
                      Algebra I
                    </div>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
